### PR TITLE
Ros package and nav2 demo

### PR DESCRIPTION
**Pull request by aaronchongth**
# 🎉 New feature

Worked off https://github.com/tfoote/ionic_demo by @tfoote

Repackages the demo world into a ROS 2 package, with basic turtlebot4 demo.

## Summary

* Massaged files into `ionic_demo` ROS 2 package, with some preliminary documentation
* Nav2 demo with a turtlebot4, together with it's laser scan of the demo map
* The current gif targets an orphan branch, https://github.com/aaronchongth/ionic_demo/tree/media, to prevent downloading large files when cloning, we can probably look into doing it for this repo too

ionic-demo-faster-smaller

## Test it

Build

```
mkdir -p ~/ionic_ws/src
cd ~/ionic_w/src
git clone https://github.com/gazebosim/ionic_demo

source /opt/ros/jazzy/setup.bash
rosdep install --from-paths src --ignore-src --rosdistro $ROS_DISTRO -y

cd ~/ionic_ws/
colcon build
```

Run the demo

```
source ~/ionic_ws/install/setup.bash
ros2 launch ionic_demo ionic_navigation_demo_launch.py headless:=0
```

On rviz, initialize the position at the origin towards the right of the map and start using the navigation commands provided on the rviz panels.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,12 +6,51 @@ Ionic demo world and resources
...
+```
+
+Launch the demo,
+
+```
+source ~/ionic_ws/install/setup.bash
+ros2 launch ionic_demo ionic_navigation_demo_launch.py headless:=0
+```
+
+On rviz, initialize the position at the origin towards the right of the map and start using the navigation commands provided on the rviz panels.
```
Comment (iche033) [inline README.md]: can you provide more instructions / examples on how to initialize the position / navigation commands in rviz panel? Or maybe there's a nav2 doc page to shows how to do this?

**Comment by iche033** [comment]
I tested this in a new container. I was able to launch gazebo + rviz2 the first time. However, when I tried to run again, gz server gives an error about not being able to find the world sdf file:

```
[ruby $(which gz) sim-6] Unable to find or download file 
[ERROR] [ruby $(which gz) sim-6]: process has died [pid 2166500, exit code 255, cmd 'ruby $(which gz) sim -r -s /tmp/nav2_qrligm4x.sdf --force-version 8'].
```

I tried switching cyclonedds but didn't help.  Any idea how to resolve this?

File: README.md (hunk 1 above)
Comment (aaronchongth) [reply README.md]: Gotcha, https://github.com/gazebosim/ionic_demo/pull/1/commits/46aef60c30f6fbda9123093b3dc635ce47e2a51d, I've added the gif and screenshots explicitly. Later on down the road we can use an orphaned media branch

**Comment by aaronchongth** [comment]
@iche033 that is a kinda odd, I've so far not experienced that. Are you running it in a distrobox? Which image are you using?
So far I'm using a fresh `ubuntu:24.04` image, with `distrobox create -i ubuntu:24.04 -n ionicdemo`, and installing ROS 2 from scratch

**Review by azeey — APPROVED** [review]
Tested with Jazzy/Harmonic and Rolling/Ionic and seems to work well.

**Pull request merged**